spec/s3.spec.js

```diff
@@ -0,0 +1,192 @@
+"use strict";
+
+describe("storage/s3", () => {
+    var AwsSdkMock, promiseMock, s3;
+    
+    /**
+     * Checks the result for the promise object passed in.
+     *
+     * @param {Object} results
+     * @param {boolean} successful
+     * @param {string} message
+     */
+    function checkPromiseResults(results, successful, message) {
+        expect(results.success).toBe(successful);
+        expect(results.value).toEqual(message);
+    }
```

Comment:
Per our discussion ... here's the code you think might be better to use because it keeps the results in with the test in a more readily accessible way.

```
expect(s3.doTheThing().status()).toEqual({
    success: true,
    value: "successful file get"
});
```